contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -0,0 +1,305 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {RouterBase, ITypeAndVersion} from "./RouterBase.sol";
+import {IFunctionsRouter} from "./interfaces/IFunctionsRouter.sol";
+import {IFunctionsCoordinator} from "./interfaces/IFunctionsCoordinator.sol";
+import {IFunctionsSubscriptions, FunctionsSubscriptions, IFunctionsBilling} from "./FunctionsSubscriptions.sol";
+
+contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions {
+  event RequestStart(
+    bytes32 indexed requestId,
+    bytes32 indexed donId,
+    uint64 indexed subscriptionId,
+    address subscriptionOwner,
+    address requestingContract,
+    address requestInitiator,
+    bytes data,
+    uint16 dataVersion,
+    uint32 callbackGasLimit
+  );
+
+  event RequestEnd(
```

Comment:
A new requirement came up from the UI - they need to display the updated subscription balance after request fulfillment (after we subtracted the total cost). Otherwise the UI/backend logic becomes very complicated to calculate the intermediate balances per each rendered request.
I suggest we simply add a parameter like `updatedSubscriptionBalance` in here, if possible.